swayfmt/src/comments.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::Write;
+use sway_ast::token::Comment;
+use sway_types::Spanned;
+
+use crate::{formatter::FormattedCode, Formatter, FormatterError};
+
+// Given a start, an end and a CommentMap, return references to all comments
+// contained within a given start and end of a span, in an exclusive range.
+pub fn get_comments_between(start: usize, end: usize, formatter: &mut Formatter) -> Vec<&Comment> {
```

Comment:
```suggestion
pub fn get_comments_between(start: usize, end: usize, formatter: &Formatter) -> Vec<&Comment> {
```
This function does not mutate the `Formatter`, so we should not take a `&mut`.

Reply:
We can't borrow the `Formatter` immutably here, since in the `Format` impls the formatter is always borrowed mutably. Is there a way to work around this?

Reply:
You can always pass a `&mut T` instance to a `&T` argument of a function, but you cannot pass a `&T` to a `&mut T` argument, as that would break Rust's ownership rules.